### PR TITLE
Update manual theme

### PR DESCRIPTION
**Pull request by CalMacCQ**
I've updated the manual to use the new documentation theme. 

Screenshot 2023-01-04 at 16 49 50

I've also done a bit of cleanup on some code snippets. Mentioned some minor new features and made the spacing more uniform. 

I've pushed all changes to `docs/manual` - hence the bloated diff. This workflow is far from ideal but necessary at present as the html is not built on the CI.

### CODE REVIEW

File: manual/conf.py [hunk 1]
```diff
@@ -15,7 +15,21 @@
...
 
-html_theme = "sphinx_rtd_theme"
+html_theme = "sphinx_book_theme"
+
+pygments_style = "pastie"
+
+html_theme = "sphinx_book_theme"
+
+html_theme_options = {
+    "repository_url": "https://github.com/CQCL/tket",
```
Comment (cqc-alec) [inline manual/conf.py:25]: What does this do? Do we want it to be this repo?

File: manual/manual_circuit.rst [hunk 2]
```diff
@@ -494,6 +515,8 @@ It is possible to specify small unitaries from ``numpy`` arrays and embed them d
     circ.add_unitary1qbox(u1box, 2)
     circ.add_unitary2qbox(u2box, 1, 0)
 
+.. note:: For performance reasons pytket currently only supports unitary synthesis up to three qubits. Three qubit synthesis can be accomplished with :py:class:`Unitary3qBox`.
```
Comment (cqc-alec) [inline manual/manual_circuit.rst:518]:
```suggestion
.. note:: For performance reasons pytket currently only supports unitary synthesis up to three qubits. Three-qubit synthesis can be accomplished with :py:class:`Unitary3qBox`.
```

File: manual/manual_circuit.rst [hunk 3]
```diff
@@ -573,25 +598,32 @@ If you are working in a Jupyter environment, a :py:class:`Circuit` can be render
 
     circ = Circuit(3)
     circ.CX(0, 1).CZ(1, 2).X(1).Rx(0.3, 0)
-    render_circuit_jupyter(circ)
+    render_circuit_jupyter(circ) # Render interactive circuit diagram
 
-``pytket`` also features ways to view the underlying DAG graphically for easier visual inspection.
+.. note:: The pytket circuit renderer can represent circuits in the standard circuit model or in the ZX Calculus representation. Other interactive features include adjustable zoom, circuit wrapping and image export.
```
Comment (cqc-alec) [inline manual/manual_circuit.rst:603]:
```suggestion
.. note:: The pytket circuit renderer can represent circuits in the standard circuit model or in the ZX representation. Other interactive features include adjustable zoom, circuit wrapping and image export. 
```

**Review by cqc-alec — CHANGES_REQUESTED** [review]

File: manual/conf.py (hunk 1 above)
Comment (CalMacCQ) [reply manual/conf.py:25]: This tells sphinx which repo to link to in the button at the top of the page. I decided to make it the tket repo as raising issues there is preferred. Although I could link to the pytket repo instead.

Comment (cqc-alec) [reply manual/conf.py:25]: OK that makes sense, thanks.

**Comment by cqc-alec** [comment]
The title "pytket-manual documentation" sounds a bit odd. Can we change it to just "pytket manual"?

**Review by cqc-alec — APPROVED** [review]

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**